tfhe/tests/backward_compatibility/shortint.rs

```diff
@@ -0,0 +1,115 @@
+        ks_base_log: DecompositionBaseLog(test_params.ks_base_log),
+        ks_level: DecompositionLevelCount(test_params.ks_level),
+        message_modulus: MessageModulus(test_params.message_modulus),
+        carry_modulus: CarryModulus(test_params.carry_modulus),
+        max_noise_level: MaxNoiseLevel::new(test_params.max_noise_level),
+        log2_p_fail: test_params.log2_p_fail,
+        ciphertext_modulus: CiphertextModulus::try_new(test_params.ciphertext_modulus).unwrap(),
+        encryption_key_choice: {
+            match &*test_params.encryption_key_choice {
+                "big" => EncryptionKeyChoice::Big,
+                "small" => EncryptionKeyChoice::Small,
+                _ => panic!("Invalid encryption key choice"),
+            }
+        },
+    }))
+}
+
+fn test_shortint_ciphertext(
+    test: ShortintCiphertextTest,
+    dir: &Path,
+) -> Result<TestSuccess, TestFailure> {
+    let key_file = dir.join(&*test.key_filename);
+    let key = ClientKey::unversionize(load_versioned(key_file).map_err(|e| test.failure(e))?)
+        .map_err(|e| test.failure(e))?;
+    let ct_file = dir.join(&*test.ct_filename);
+    let ct = Ciphertext::unversionize(load_versioned(ct_file).map_err(|e| test.failure(e))?)
+        .map_err(|e| test.failure(e))?;
+
+    let clear = key.decrypt(&ct);
+    if clear != test.clear_value {
+        Err(test.failure(format!(
+            "Invalid decrypted cleartext:\n Expected :\n{:?}\nGot:\n{:?}",
+            clear, test.clear_value
+        )))
+    } else {
+        Ok(test.success())
+    }
+}
+
+fn test_shortint_clientkey(
+    test: ShortintClientKeyTest,
+    dir: &Path,
+) -> Result<TestSuccess, TestFailure> {
+    let versioned_key =
+        load_versioned(dir.join(&*test.key_filename)).map_err(|e| test.failure(e))?;
+
+    let test_params = load_params(&test.parameters);
+
+    let key = ClientKey::unversionize(versioned_key).map_err(|e| test.failure(e))?;
+
+    if test_params != key.parameters {
+        Err(test.failure(format!(
+            "Invalid parameters:\n Expected :\n{:?}\nGot:\n{:?}",
+            test_params, key.parameters
+        )))
+    } else {
+        Ok(test.success())
+    }
+}
+
+pub fn test_shortint<P: AsRef<Path>>(base_dir: P) -> Vec<Result<TestSuccess, TestFailure>> {
+    let shortint_dir = base_dir.as_ref().join("shortint");
+
+    let meta = load_metadata(shortint_dir.join("shortint.ron")).unwrap();
+
+    let mut results = Vec::new();
+
+    for testcase in meta {
+        let test_result = match testcase {
+            TestMetadata::ShortintCiphertext(test) => test_shortint_ciphertext(test, &shortint_dir),
+            TestMetadata::ShortintClientKey(test) => test_shortint_clientkey(test, &shortint_dir),
+        };
+
+        match &test_result {
+            Ok(r) => println!("{}", r),
+            Err(r) => println!("{}", r),
+        }
+        results.push(test_result)
+    }
+
+    results
+}
```

Comment:
how do the tests know to look in 0.6 and move up ?

Reply:
I think the right way would be to have the ron metadata files at a fixed location in the data folder (not the case at the moment). The metadata for a test holds the path to the cbor serialized case so this is what should be updated. We could also add a version number in the testcase metadata to be able to skip testcases for versions that are too recent

Reply:
done: the ron file is now loaded from the root of the data directory. Each testcase now comes with additional metadata, the "version" and the "module" which indicates the subfolder where the serialized files are stored.